src/Sulu/Component/Security/Authorization/SecurityContextVoter.php

```diff
@@ -0,0 +1,223 @@
+<?php
+/*
+ * This file is part of the Sulu CMS.
+ *
+ * (c) MASSIVE ART WebServices GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Component\Security\Authorization;
+
+use Doctrine\Common\Collections\Collection;
+use Sulu\Bundle\SecurityBundle\Entity\Group;
+use Sulu\Bundle\SecurityBundle\Entity\Permission;
+use Sulu\Bundle\SecurityBundle\Entity\Role;
+use Sulu\Bundle\SecurityBundle\Entity\User;
+use Sulu\Bundle\SecurityBundle\Entity\UserGroup;
+use Sulu\Bundle\SecurityBundle\Entity\UserRole;
+use Symfony\Component\Security\Acl\Domain\ObjectIdentity;
+use Symfony\Component\Security\Acl\Exception\AclNotFoundException;
+use Symfony\Component\Security\Acl\Model\AclProviderInterface;
+use Symfony\Component\Security\Acl\Model\ObjectIdentityInterface;
+use Symfony\Component\Security\Core\Authentication\Token\TokenInterface;
+use Symfony\Component\Security\Core\Authorization\Voter\VoterInterface;
+
+class SecurityContextVoter implements VoterInterface
+{
+    /**
+     * The permissions available, defined by config
+     * @var array
+     */
+    private $permissions;
+
+    /**
+     * @var AclProviderInterface
+     */
+    private $aclProvider;
+
+    public function __construct($permissions, AclProviderInterface $aclProvider)
+    {
+        $this->permissions = $permissions;
+        $this->aclProvider = $aclProvider;
+    }
+
+    /**
+     * Checks if the voter supports the given attribute.
+     *
+     * @param string $attribute An attribute
+     * @return Boolean true if this Voter supports the attribute, false otherwise
+     */
+    public function supportsAttribute($attribute)
+    {
+        return in_array($attribute, array_keys($this->permissions));
+    }
+
+    /**
+     * Checks if the voter supports the given class.
+     *
+     * @param string $class A class name
```

Comment:
We should standardize on having an empty line or not having an empty line before `@params`. I would vote for an empty line as this is the defacto standard in Symfony